scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -18,24 +18,43 @@
 
 package io.ballerina.scan.internal;
 
+import io.ballerina.compiler.syntax.tree.CheckExpressionNode;
 import io.ballerina.compiler.syntax.tree.ModulePartNode;
 import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
 import io.ballerina.compiler.syntax.tree.SyntaxTree;
 import io.ballerina.projects.Document;
+import io.ballerina.scan.ScannerContext;
 
 /**
  * {@code StaticCodeAnalyzer} contains the logic to perform core static code analysis on Ballerina documents.
  *
  * @since 0.1.0
  * */
 class StaticCodeAnalyzer extends NodeVisitor {
+    private final Document document;
     private final SyntaxTree syntaxTree;
+    private final ScannerContext scannerContext;
 
     StaticCodeAnalyzer(Document document, ScannerContextImpl scannerContext) {
+        this.document = document;
         this.syntaxTree = document.syntaxTree();
+        this.scannerContext = scannerContext;
     }
 
     void analyze() {
         this.visit((ModulePartNode) syntaxTree.rootNode());
     }
+
+    /**
+     * Visits check expressions in a Ballerina document and perform static code analysis.
+     *
+     * @param checkExpressionNode node that represents a check expression
+     */
+    @Override
+    public void visit(CheckExpressionNode checkExpressionNode) {
+        if (checkExpressionNode.checkKeyword().kind().equals(SyntaxKind.CHECKPANIC_KEYWORD)) {
+            scannerContext.getReporter().reportIssue(document, checkExpressionNode.location(), 1);
+        }
```

Comment:
For each core rule, I would have a final variable in the CoreRules class and use the pass by Rule interface here. Improves readability.

At least, the rule ID has to be a constant like that if we can't make the rule a constant.